python/tank/bootstrap/manager.py

```diff
@@ -326,14 +333,37 @@ def bootstrap_engine(self,
             self._bootstrapper.set_callbacks(progress_callback, completed_callback, failed_callback)
             self._bootstrapper.bootstrap()
 
-            # At this point, the engine has not yet been started.
-            engine = None
-
         else:
 
-            # Execute the whole engine bootstrap logic synchronously in the main application thread.
-            tk = self.bootstrap_sgtk(engine_name, entity)
-            engine = self.start_engine(tk, engine_name, entity)
+            bootstrap_step = None
+
+            try:
+
+                # Install the bootstrap progress reporting callback.
+                self.set_progress_callback(progress_callback)
+
+                # Execute the whole engine bootstrap logic synchronously in the main application thread.
+                bootstrap_step = "sgtk"
```

Comment:
Maybe we don't need to define this? You could wrap each of these calls into their own try/except and raise accordingly. For the values sent to the callback, I'd consider making these class attributes: `ToolkitManager.TK_BOOTSTRAP_PHASE`, `ToolkitManager.TK_ENGINE_STARTUP_PHASE`... rather than hardcoding here.